[CUDA] Build issue with sycl-trace under shared-libs

### Describe the bug

While building the project, I see this:

Configured using:
```
python3 ./buildbot/configure.py \
  -o build_cuda${cuda_version}_${NOW} \
  -t Release \
  --native_cpu \
  --use-lld \
  --llvm-external-projects lld,mlir,flang \
  --cuda -DCUDA_Toolkit_ROOT=/soft/compilers/cuda/cuda-$cuda_version \
  --cmake-opt="-DLLVM_ENABLE_RUNTIMES=openmp;compiler-rt;flang-rt" \
  --cmake-opt="-DLLVM_ENABLE_PROJECTS=clang;sycl;llvm-spirv;opencl;xpti;xptifw;libdevice;sycl-jit;lld;mlir;flang;libclc" \
  --cmake-opt="-DLLVM_ENABLE_LLD=OFF" \
  --cmake-opt="-DLLVM_TOOL_LLD_BUILD=ON" \
  --cmake-opt="-DCMAKE_C_COMPILER=gcc" \
  --cmake-opt="-DCMAKE_CXX_COMPILER=g++"
```

Error:

```
FAILED: tools/sycl/tools/sycl-trace/CMakeFiles/cuda_trace_collector.dir/cuda_trace_collector.cpp.o 
/soft/compilers/gcc/13.3.0/x86_64-suse-linux/bin/g++ -DUSE_UR_CUDA -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_GLIBCXX_USE_CXX11_ABI=1 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Dcuda_trace_collector_EXPORTS -I/vast/users/abagusetty/compilers/llvm/build_cuda12.9.1_12-19-2025/tools/sycl/tools/sycl-trace -I/vast/users/abagusetty/compilers/llvm/sycl/tools/sycl-trace -I/vast/users/abagusetty/compilers/llvm/build_cuda12.9.1_12-19-2025/include -I/vast/users/abagusetty/compilers/llvm/llvm/include -I/vast/users/abagusetty/compilers/llvm/xptifw/include -I/vast/users/abagusetty/compilers/llvm/build_cuda12.9.1_12-19-2025/_deps/parallel-hashmap-src -I/vast/users/abagusetty/compilers/llvm/xpti/../xpti/include -isystem /soft/compilers/cuda/cuda-12.9.1/extras/CUPTI/include -isystem /soft/compilers/cuda/cuda-12.9.1/include -isystem /vast/users/abagusetty/compilers/llvm/build_cuda12.9.1_12-19-2025/_deps/emhash-src -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -fno-lifetime-dse -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-uninitialized -Wno-nonnull -Wno-class-memaccess -Wno-dangling-reference -Wno-redundant-move -Wno-pessimizing-move -Wno-array-bounds -Wno-stringop-overread -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -Wall -Wextra -O3 -DNDEBUG -std=c++17 -fPIC -UNDEBUG -MD -MT tools/sycl/tools/sycl-trace/CMakeFiles/cuda_trace_collector.dir/cuda_trace_collector.cpp.o -MF tools/sycl/tools/sycl-trace/CMakeFiles/cuda_trace_collector.dir/cuda_trace_collector.cpp.o.d -o tools/sycl/tools/sycl-trace/CMakeFiles/cuda_trace_collector.dir/cuda_trace_collector.cpp.o -c /vast/users/abagusetty/compilers/llvm/sycl/tools/sycl-trace/cuda_trace_collector.cpp
/vast/users/abagusetty/compilers/llvm/sycl/tools/sycl-trace/cuda_trace_collector.cpp:14:10: fatal error: sycl/detail/spinlock.hpp: No such file or directory
   14 | #include <sycl/detail/spinlock.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

### To reproduce

_No response_

### Environment

- OS: Linux
- Target device and vendor: Nvidia

### Additional context

_No response_

--- a/sycl/tools/sycl-trace/CMakeLists.txt
+++ b/sycl/tools/sycl-trace/CMakeLists.txt
@@ -32,6 +32,9 @@ if ("cuda" IN_LIST SYCL_ENABLE_BACKENDS)
     cuda_trace_collector
   )
   add_dependencies(sycl-trace cuda_trace_collector)
+  target_include_directories(cuda_trace_collector PRIVATE
+    "${sycl_inc_dir}"
+  )
 endif()
 
 add_library(sycl_ur_trace_collector SHARED

